devtools/aesh/src/main/java/io/quarkus/cli/commands/ExecuteUtil.java

```diff
@@ -0,0 +1,106 @@
+import java.util.stream.Stream;
+
+import org.aesh.command.invocation.CommandInvocation;
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.InvocationResult;
+import org.apache.maven.shared.invoker.Invoker;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+
+public class ExecuteUtil {
+
+    public static void executeGradle(File projectDirectory, CommandInvocation invocation, String buildTarget) {
+        String gradleExecutable = findExecutable("gradle");
+        if (gradleExecutable == null) {
+            invocation.println("unable to find the gradle executable, is it in your path?");
+        } else {
+            gradleExecutable += File.separator + "bin" + File.separator + "gradle";
+
+            try {
+                Process process = new ProcessBuilder()
+                        .command(gradleExecutable, buildTarget)
+                        .directory(projectDirectory)
+                        .start();
+
+                BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    invocation.println(line);
+                }
+
+                int exit = process.waitFor();
+                if (exit != 0)
+                    invocation.println("Build failed.");
+            } catch (IOException | InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public static void executeMaven(File projectDirectory, CommandInvocation invocation, String buildTarget) {
+        String mvnPath = findExecutable("mvn");
+        System.setProperty("maven.home", mvnPath);
+
+        InvocationRequest request = new DefaultInvocationRequest();
+        request.setPomFile(new File(projectDirectory.getAbsolutePath() + File.separatorChar + "pom.xml"));
+        request.setGoals(Collections.singletonList(buildTarget));
+
+        Invoker invoker = new DefaultInvoker();
+
+        InvocationResult result = null;
+        try {
+            result = invoker.execute(request);
+        } catch (MavenInvocationException e) {
+            e.printStackTrace();
+        }
+
+        if (result.getExitCode() != 0) {
+            invocation.println("Build failed.");
+        }
+    }
+
+    public static String findExecutable(String exec) {
+        Optional<Path> mvnPath = Stream.of(System.getenv("PATH").split(Pattern.quote(File.pathSeparator)))
+                .map(Paths::get)
+                .filter(path -> Files.exists(path.resolve(exec))).findFirst();
+
+        return mvnPath.map(value -> value.getParent().toString()).orElse(null);
+    }
+
+    public static void executeWrapper(CommandInvocation invocation, File wrapper, String target) {
+        try {
```

Comment:
why is it that you can't just use the same way of running the wrapper and the path executable ?

Reply:
If we detect a maven install we use the maven-invocation api to kick off the build. That is what I'll try to do with the gradle-api as well, but since they are called differently now I had to distinct wrapper and executable.